core/issuers.chainmail

```diff
@@ -1,56 +1,174 @@
+  getIssuer() -> (Issuer);
+
+  /**
+   * Create a new Purse containing the amount. Give it the specified name or "a
+   * purse".
+   */
+  mint(initialBalance:Amount, name:String = 'a purse') -> (Purse);
+
+  /** Reset all purses to contain nothing. */
+  destroyAll();
+
+  /** (Non-fungible amounts only.) Ensure no purse contains amount. */
+  destroy(amount:Amount);
+
+  /**
+   * Destroy any purses or payments containing amount, then issue a new Purse
+   * with that amount.
+   */
+  revoke(amount:Amount) -> (Purse);
+}
+
+/**
+ * Purses hold verified amounts of certain rights issued by Mints. Purses can
+ * transfer part of the balance they hold in a Payment, which has a narrower
+ * interface. A purse's balance can rise and fall, through the action of deposit
+ * and withdraw, while payments don't accept deposits.
+ *
+ * The primary use for Purses and Payments is for currency-like and goods-like
+ * valuables, but they can also be used to represent other kinds of rights, such
+ * as the right to participate in a particular contract.
+ *
+ * Currency-like assets are fungible, and don't distinguish use rights from
+ * transfer rights. The only use of a currency is to spend it. Some goods
+ * distinguish use rights from transfer rights, like a membership card; you can
+ * exercise the rights without giving them up.
+ */
+interface Purse extends (PurseOrPayment) {
+  /** Get the Issuer for this mint. */
+  getIssuer() -> (Issuer);
+  getXferBalance() -> (Amount);
+  getUseBalance() -> (Amount);
+  deposit(amount:Amount, srcPayment:PurseOrPayment) -> (Amount);
+  depositAll(srcPayment:PurseOrPayment) -> (Amount);
+  withdraw(amount:Amount, name:String) -> (Payment);
+  withdrawAll(name:String) -> (Payment);
 }
-makeNatAssay(label ::Label) ::Assay;
-makeMetaSingleAssayMaker(
-  baseLabelToAssayFn ::(Label -> Assay)) ::(Label -> Assay);
-
-class Issuer[Assay] {
-  type Amount = Assay.Amount;
-  getLabel() ::{ issuer ::Issuer, description };
-  getAssay() ::Assay;
-  makeEmptyPurse(name ?String) ::Purse;
-
-  class Mint {
-    getIssuer() Issuer;
-    mint(initialBalance ?Amount, name ?String) ::Purse;
-  }
-  class Payment {
-    getIssuer() ::Issuer;
-    getXferBalance() ::Amount;
-  }
-  class Purse {
-    getIssuer() ::Issuer;
-    getXferBalance() ::Amount;
-    getUseBalance() ::Amount;
-    deposit(amount ?Amount, srcPaymentP ?reveal[Promise]) ::Amount;
-    withdraw(amount ?Amount, name ?String) ::Purse;
-  }
+
+/**
+ * Payments hold verified amounts of certain rights issued by Mints. Amounts
+ * from payments can be deposited in purses, but otherwise, the entire amount is
+ * available when the payment is transferred. A payment's balance can only fall,
+ * through the action of withdraw.
+ * 
+ * Payments can be converted to Purses by calling
+ * payment.getIssuer().makeEmptyPurse().depositAll(payment);
+ */
+interface Payment extends (PurseOrPayment) {
+  getIssuer() -> (Issuer);
+  getXferBalance() -> (Amount);
```

Comment:
only getBalance() now

Reply:
I'll write it up when it becomes stable.

Reply:
Hmm, why do you think that this part is not stable?

Reply:
PurseOrPayment was still in flux. Now that that's gone, I've moved to the new method names and added doc.